[STLND-1950] - Version 0.1.2

jaszczur171 opened the pull request:
- Bumped version to 0.1.2
- Added event wrapper as top level wrapper for messages. This now allows composing messages from raw JSON.
- Decomposed message classes to its basic components, giving more flexibility to the code.
- Created StellaNowBaseModel that implements converters at a base level of every message for all of its members.
- Used pydantic exclude feature to remove relevant fields during message serialisation
- Utilised pydantic serialisation alias to rename fields on the fly during serialisation
- Updated wrapper tests with the new approach

jaszczur171 committed:
[STLND-1950] - Version 0.1.2
- Bumped version to 0.1.2
- Added event wrapper as top level wrapper for messages. This now allows composing messages from raw JSON.
- Decomposed message classes to its basic components, giving more flexibility to the code.
- Created StellaNowBaseModel that implements converters at a base level of every message for all of its members.
- Used pydantic exclude feature to remove relevant fields during message serialisation
- Utilised pydantic serialisation alias to rename fields on the fly during serialisation
- Updated wrapper tests with the new approach

jaszczur171 committed:
[STLND-1950] - Version 0.1.2
- Bumped version to 0.1.2
- Added event wrapper as top level wrapper for messages. This now allows composing messages from raw JSON.
- Decomposed message classes to its basic components, giving more flexibility to the code.
- Created StellaNowBaseModel that implements converters at a base level of every message for all of its members.
- Used pydantic exclude feature to remove relevant fields during message serialisation
- Utilised pydantic serialisation alias to rename fields on the fly during serialisation
- Updated wrapper tests with the new approach

jaszczur171 committed:
[STLND-1950] - Version 0.1.2
- Fixed datetime formatting

jaszczur171 committed:
[STLND-1950] - Version 0.1.2
- Removed leftover print in a test

sebtom reviewed (commented):
LGTM, but it's still draft so not approving

jaszczur171 committed:
[STLND-1950] - Version 0.1.2
- Added new message classes generated by updated CLI in version 0.0.16
- Added version check in pull request to develop to see if version already exists.If so it will be rejected, forcing the developer to increment version
- Lint fixes

jaszczur171 committed:
[STLND-1950] - Version 0.1.2
- Added new message classes generated by updated CLI in version 0.0.16
- Added version check in pull request to develop to see if version already exists.If so it will be rejected, forcing the developer to increment version
- Adjusted the whole codebase to work with event wrappers instead of message wrappers
- Added ability to send messages with pure JSON as payload by using the StellaNowMessageWrapper
- Updated demo app to how the sdk operates now
- Lint fixes

MaciekKrz commented on combined_files.txt:
I think this file should be removed :)

jaszczur171 replied:
damn ... it should

MaciekKrz commented on stellanow_sdk_python_demo/main.py:
```diff
@@ -41,25 +41,34 @@
+    # sdk = configure_dev_username_password_mqtt_lifo_sdk()
     shutdown_event = asyncio.Event()  # Event to signal shutdown
 
+    # TestMessage(
+    #     phone_number=PhoneNumber(country_code=48, number=700000),
+    #     user_id="user_98888",
+    #     event_name="test_event",
+    #     entities=[Entity(entity_type_definition_id="test", entity_id="test_id")]
+    # )
```
Let's remove the commented code.

MaciekKrz commented on stellanow_sdk_python_demo/main.py:
```diff
@@ -41,25 +41,34 @@
         await sdk.start()
         for i in range(10):
-            message = UserDetailsMessage(
+            logger.info(f"I am here {i}")
```
This logger should be removed as well.

MaciekKrz commented on stellanow_sdk_python_demo/models/__init__.py:
__init__.py removed but please remove also folder.

jaszczur171 replied:
I am not sire what you mean? 

Since python version 3.3 the __init__ file is implicit, and as such it is not required. We target 3.11 as python version, so for all intense and purposes we do not need to satisfy anything so outdated as 3.3.

MaciekKrz replied:
Ok, we misunderstand. It's an empty models folder, but it will be deleted after the merge.

jaszczur171 committed:
[STLND-1950] - Version 0.1.2
- Removed some leftover debug / development related code

jaszczur171 committed:
[STLND-1950] - Version 0.1.2
- Removed code dump file

MaciekKrz reviewed (approved):
LGTM
